wandb/util.py

```diff
@@ -1859,7 +1859,7 @@ def sample_with_exponential_decay_weights(
     sampled_indices = np.random.choice(len(xs_array), size=sample_size, p=weights)
     sampled_xs = xs_array[sampled_indices].tolist()
     sampled_ys = ys_array[sampled_indices].tolist()
-    sampled_keys = keys_array[sampled_indices].tolist() if keys else None
+    sampled_keys = keys_array[sampled_indices].tolist() if keys_array else None
```

Comment:
lol this was just wrong

Reply:
technically, if `keys` is truthy, that means `keys_array` is `np.array(keys)` (see a few lines above) and therefore an `ndarray`; but you can't rely on mypy figuring that out

Reply:
ah, i didn't expend the code beyond this line, so i didn't see that, but in any case it still seems more correct to actually check `keys_array`